### PR TITLE
feat: tag component 

### PR DESCRIPTION
**Pull request by kylemcd**
### Description
`@telegraph/tag`
- Adds component
- Adds tests

`@telegraph/playground`
- Adds tag component examples

`@telegraph/button`
- Adds new button colors to be used in tag component

`@telegraph/tokens`
- Fixes issue with yellow tokens being incorrect

### Screenshot
Screenshot 2024-03-04 at 11 21 22 AM

### CODE REVIEW

**Review by thomaswhyyou — COMMENTED** [review]
Have a couple small questions, but overall this makes sense to me!

File: packages/tag/src/Tag/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  size?: "1" | "2";
+  color?: React.ComponentProps<typeof TelegraphButton>["color"] & "default";
+  variant?: "soft" | "solid";
+};
+
+type RootRef = HTMLSpanElement;
+
+const TagContext = React.createContext<RootProps>({
+  size: "1",
+  color: "default",
```
[the remark this answers is not in the document]
Comment (kylemcd) [reply packages/tag/src/Tag/Tag.tsx:23]: I think it's an issue with the design system button, going to talk to Krisna about it.

File: packages/tag/src/Tag/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        className={clsx("rounded-tl-0 rounded-bl-0 mr-1", className)}
+        {...props}
+        ref={forwardedRef}
+      />
+    );
+  },
+);
+
+type DefaultProps = React.ComponentProps<typeof Root> & {
+  text: string;
```
[the remark this answers is not in the document]
Comment (kylemcd) [reply packages/tag/src/Tag/Tag.tsx:111]: It could be done either way, I can change it to use children.

[the remark this answers is not in the document]
Comment (thomaswhyyou) [reply packages/tag/src/Tag/Tag.tsx:111]: No need to change if you are happy with! No strong opinion but was just curious. We can also use it as is and see how it feels ergonomically, and can always revisit.

File: packages/tag/src/Tag/Tag.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ksorathia) [reply packages/tag/src/Tag/Tag.tsx:23]: It's better to think about Tags as a type of label or data display, so it's closer to Text/Heading than it is to a Button.

This is why we have a "default" tag (for default Text), and a "gray" tag (for gray Text).

Buttons, instead, follow our color scale a bit closer, as they are meant to be visually styled to pair with our theme colors.

[the remark this answers is not in the document]
Comment (thomaswhyyou) [reply packages/tag/src/Tag/Tag.tsx:23]: Ah okay, I didn't realize a "default" tag and a "gray" tags render two different things!

**Pull request merged**